### PR TITLE
[WIP] Integrate Localhost's data model

### PR DESCRIPTION
**Pull request by bscofield**

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -226,3 +238,78 @@ def as_geojson(self):
             },
             "geometry": mapping(to_shape(self.point))
         }
+
+class Contact(db.Model, UuidMixin):
```
[the remark this answers is not in the document]
Comment (bscofield) [reply app/models.py]: Bringing convo from Slack here:

I'm pretty sure the way Localhost currently works is that the dates are set in `Visit` - `start_date` and `end_date`

I agree that `Contact` needs the state, though -- that way I can set up a `Visit` (I'm going to Chicago on these dates) and reach out to various hosts (multiple `Contact`s), of which only one gets accepted

Does that make sense?

[the remark this answers is not in the document]
Comment (bscofield) [reply app/models.py]: Resolved, breaking with Localhost and adding dates to `Contact`

**Comment by bscofield** [comment]
OK, added new fields to Contact -- and I went ahead and removed the note about adding more methods for the new tables, since I'm not sure what we'll need as an API once the new user flows are settled. Any other comments @samuelcole @iandees?

**Comment by iandees** [comment]
Sorry for the late reply.

It's not clear to me what these models mean in a general sense. Is there a doc in Localhost somewhere that describes user flows and how these models relate to them?

**Comment by bscofield** [comment]
@iandees I haven't seen such a document, but here's my understanding (@samuelcole, please correct if I'm wrong)

* a *Visit* is a trip being planned by a traveler -- i.e., I'm going to canvass in PA in the last week of March
* a *Hosting* is a bed / room / whatever being offered by a host -- i.e., my guest room is available 1-7 June
* a *Contact* is a request from a traveler to a host to occupy their space -- "hey, can I get that spare room for Saturday the 12th?"

**Comment by samuelcole** [comment]
`visit` is a bad/vague name in localhost, but it's a request to stay near a lat/long.

`hosting` is an available place to sleep

`contact` is contact between a 'visitor' and a 'host', i.e. the request to occupy that space

so yeah, what @bscofield said

**Pull request closed without merging**